plugins/ftdidmx/FtdiWidget.cpp

```diff
@@ -364,15 +372,35 @@ bool FtdiInterface::Write(const ola::DmxBuffer& data) {
   }
 }
 
-bool FtdiInterface::Read(unsigned char *buff, int size) {
-  int read = ftdi_read_data(&m_handle, buff, size);
-  if (read <= 0) {
+
+bool FtdiInterface::Write(ola::io::ByteString *packet) {
+  int bytesWritten = ftdi_write_data(
+                       &m_handle,
+                       const_cast<unsigned char *>(packet->data()),
+                       packet->size());
+
+  int size = packet->size();
+  if (bytesWritten < 0) {
     OLA_WARN << m_parent->Description() << " "
              << ftdi_get_error_string(&m_handle);
     return false;
-  } else {
+  } else if (bytesWritten == static_cast<int>(packet->size())) {
```

Comment:
It might be clearer to have all the good or bad cases together, rather than bad, good else bad. I'm guessing you've possibly put < 0 first as an optimisation, as it only needs one side, but perhaps doing < 0, != else true would be better?

Reply:
You are absolutely right, changed.